### PR TITLE
Feature/condo/sberdoma 306/prevent user to leave property form without saving

### PR DESCRIPTION
**Pull request by dkoviazin**
1. Didn't found any legal way to prevent "next/router" from navigation, so used the suggested hack and added todo
2. Added "sberDanger" button
Alert

### CODE REVIEW

File: apps/condo/domains/property/components/BasePropertyForm/index.tsx [hunk 1]
```diff
@@ -20,20 +22,59 @@ interface IPropertyFormProps {
...
+    const closeModal = () => {
+        setIsMapChanged(false)
+        setIsModalVisible(false)
+    }
+    const handleCancel = () => {
+        closeModal()
+        router.push(next)
+    }
+
+    useEffect(() => {
```
Comment (Dimitreee) [inline apps/condo/domains/property/components/BasePropertyForm/index.tsx:55]: Please, create hook/component for common usage with all forms.

Comment (dkoviazin) [reply apps/condo/domains/property/components/BasePropertyForm/index.tsx]: Done

**Review by Dimitreee — APPROVED** [review]

File: apps/condo/domains/common/components/Prompt.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    form: FormInstance<unknown>
+    handleSave: () => void
+}
+
+const Prompt: React.FC<IPromptProps> = ({ children, title, form, handleSave: formSubmit }) => {
+    const [next, setNext] = useState(null)
+    const isIgnoringPrompt = useRef(false)
+    const [isModalVisible, setIsModalVisible] = useState(false)
+    const router = useRouter()
+    const intl = useIntl()
```
Comment (pahaz) [inline apps/condo/domains/common/components/Prompt.tsx:18]: It's better to use common style for components: https://github.com/open-condo-software/condo/blob/master/docs/style.md#react-component-style

intl should be the first

File: apps/condo/domains/property/components/BasePropertyForm/index.tsx [hunk 3]
```diff
@@ -20,11 +22,15 @@ interface IPropertyFormProps {
     type: string
 }
 
-export const BasePropertyForm: React.FC<IPropertyFormProps> = (props) => {
+const BasePropertyForm: React.FC<IPropertyFormProps> = (props) => {
     const intl = useIntl()
     const AddressLabel = intl.formatMessage({ id: 'pages.condo.property.field.Address' })
     const FieldIsRequiredMsg = intl.formatMessage({ id: 'FieldIsRequired' })
     const NameMsg = intl.formatMessage({ id: 'pages.condo.property.form.field.Name' })
+
```
Comment (pahaz) [inline apps/condo/domains/property/components/BasePropertyForm/index.tsx:30]:
```suggestion
```

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/common/components/Prompt.tsx (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/common/components/Prompt.tsx:20]: Done

File: apps/condo/domains/property/components/BasePropertyForm/index.tsx (hunk 3 above)
Comment (dkoviazin) [reply apps/condo/domains/property/components/BasePropertyForm/index.tsx:30]: Done

**Pull request merged**